### PR TITLE
Discontinue deprecated way of specifying kafka logs dir

### PR DESCRIPTION
**Pull request by nsharma-git**
# Description

Discontinue the deprecated way of providing/overriding the data directory for kafka. Going forward, we should be using 
kafka_broker_custom_properties for the same. 
```
kafka_broker_custom_properties:
    log.dirs: dir1,dir2
```
Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
`molecule --debug converge -s mtls-ubuntu `

**Test Configuration**:
Docker based molecule scenario.
```
PLAY RECAP *********************************************************************
control-center1            : ok=78   changed=31   unreachable=0    failed=0    skipped=40   rescued=0    ignored=0
kafka-broker1              : ok=83   changed=36   unreachable=0    failed=0    skipped=48   rescued=0    ignored=0
kafka-broker2              : ok=83   changed=36   unreachable=0    failed=0    skipped=45   rescued=0    ignored=0
kafka-broker3              : ok=83   changed=36   unreachable=0    failed=0    skipped=45   rescued=0    ignored=0
kafka-connect1             : ok=78   changed=32   unreachable=0    failed=0    skipped=49   rescued=0    ignored=0
kafka-rest1                : ok=75   changed=30   unreachable=0    failed=0    skipped=43   rescued=0    ignored=0
ksql1                      : ok=74   changed=31   unreachable=0    failed=0    skipped=43   rescued=0    ignored=0
schema-registry1           : ok=72   changed=30   unreachable=0    failed=0    skipped=43   rescued=0    ignored=0
zookeeper1                 : ok=81   changed=31   unreachable=0    failed=0    skipped=49   rescued=0    ignored=0

```
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/variables/defaults/main.yml [hunk 1]
```diff
@@ -433,10 +433,6 @@ kafka_broker:
   appender_log_path: "{{kafka_broker_default_log_dir}}"
   # Deprecated way of providing custom properties
   properties: {}
```
Comment (utkarsh5474) [inline roles/variables/defaults/main.yml:435]: Can this also be deprecated since we are using kafka_broker_custom_properties now?

**Comment by utkarsh5474** [comment]
Also, can we point this to earlier branches? Code from 6.0.x onwards has this comment of deprecation.

**Comment by nsharma-git** [comment]
> Also, can we point this to earlier branches? Code from 6.0.x onwards has this comment of deprecation.

Typically we keep the deprecated functionality for a one major version so that users have sufficient time to migrate.

File: roles/variables/defaults/main.yml (hunk 1 above)
Comment (nsharma-git) [reply roles/variables/defaults/main.yml:435]: We should. Probably a different PR would be better.

**Review by utkarsh5474 — APPROVED** [review]
LGTM, thanks

**Pull request merged**